cwltool/singularity.py

```diff
@@ -370,6 +370,12 @@ def add_writable_directory_volume(
                 self.append_volume(runtime, source, volume.target, writable=True)
                 ensure_writable(source)
 
+    def _required_env(self) -> Dict[str, str]:
+        return {
+            "TMPDIR": self.CONTAINER_TMPDIR,
+            "HOME": self.builder.outdir,
```

Comment:
`WARNING: Overriding HOME environment variable with SINGULARITYENV_HOME is not permitted`

https://github.com/common-workflow-language/cwltool/pull/1378/checks?check_run_id=2237763816#step:9:100

```suggestion
            "SINGULARITYENV_HOME"": self.builder.outdir,
```